File: tourneyAPI/Models/Entities/Characters/Sonic.cs
namespace Entities;

using Enums;


// Defines the competitive profile metadata for this playable character.
public class Sonic : Character
{
    // Initializes this character's default competitive attributes for matchmaking and tier logic.
    public Sonic()
    {
        Id = CharacterId.Sonic;
        characterName = CharacterName.SONIC;
        archetype = Archetype.HIT_AND_RUN;
        fallSpeed = FallSpeed.FAST_FALLERS;
        weightClass = WeightClass.LIGHTWEIGHT;
        tierPlacement = TierPlacement.S;
    }

}

File: tourneyClient/src/models/entities/Characters/Sonic.ts
import { CharacterId } from "@/models/Enums/CharacterId";
import { Character } from "../Character"
import { CharacterName } from "@/models/Enums/CharacterName";
import { Archetype } from "@/models/Enums/Archetype";
import { FallSpeed } from "@/models/Enums/FallSpeed";
import { WeightClass } from "@/models/Enums/WeightClass";
import { TierPlacement } from "@/models/Enums/TierPlacement";

export const Sonic: Character =
{
    id: CharacterId.Sonic,
    characterName: CharacterName.SONIC,
    archetype: Archetype.HIT_AND_RUN,
    fallSpeed: FallSpeed.FAST_FALLERS,
    weightClass: WeightClass.LIGHTWEIGHT,
    tierPlacement: TierPlacement.S
}

export default Sonic; 